hw/hal/include/hal/hal_system.h

```diff
@@ -72,6 +72,10 @@ enum hal_reset_reason {
     HAL_RESET_SYS_OFF_INT = 7,
     /** Restart due to DFU */
     HAL_RESET_DFU = 8,
+    /** Reserved 1 */
+    HAL_RESET_WATCHDOG2 = 9,
```

Comment:
this file is common for all MCUs and imo it does not really need to have unique mapping for every single reset reason for each MCU. in this case, CMAC watchdog could be just mapped to `HAL_RESET_WATCHDOG`. in generic case, it would be just better to add `HAL_RESET_OTHER` and map any reset reason that does not map to defined value - MCU can provide `mcu_reset_cause` to read actual reset reason and appropriate `#define`s in `mcu.h`.